Bugzilla/Search.pm

```diff
@@ -2371,6 +2374,19 @@ sub SqlifyDate {
         return sprintf("%4d-%02d-01 00:00:00", $year + 1900, $month + 1);
       }
       else {
+        # If day is 31 but month only has 30 days, then set day to 30
```

Comment:
It would be better to use a library that performs correct date math rather than hand rolling, especially when it comes to handling leap years.

I'd also expect dates that overflow months to map to the following month; eg. 2022-11-31 -> 2022-12-01